classes/form/edit_experiment.php

```diff
@@ -0,0 +1,87 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Form for editing experiments
+ *
+ * @package   tool_abconfig
+ * @author    Peter Burnett <peterburnett@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace tool_abconfig\form;
+
+defined('MOODLE_INTERNAL') || die();
+require_once("$CFG->libdir/formslib.php");
+
+class edit_experiment extends \moodleform {
+
+    public function definition() {
+        $mform = $this->_form;
+
+        // Hidden form element for experiment id
+        $mform->addElement('hidden', 'shortname', $this->_customdata['shortname']);
+        $mform->setType('shortname', PARAM_TEXT);
+
+        // Experiment data
+        $name = $this->_customdata['name'];
```

Comment:
These 6 lines with customdata seem redundant, there should be a $mform->set_data($toform); call on the page which creates the form object which does all this in a single hit